### PR TITLE
11633 fix html validation errors on blog page

### PR DESCRIPTION
**Pull request by tbrlpld**
# Description

This PR fixes 2 HTML validation errors and 6 template errors on the blog page. 

The HTML validation errors are fixed by dropping the duplicate `share-buttons.html` template and reusing the one that was already fixed when fixing issues on the PNI article page. The kept template is also moved to the more central template location. 

All other HTML validation errors still present in production have already been fixed in previous PRs.

Link to sample test page: https://foundation.mozilla.org/en/blog/how-to-tell-chat-gpt-generated-text/
Related PRs/issues: #11633

# Checklist

**Tests**
- [-] Is the code I'm adding covered by tests?

**Changes in Models:**
- [-] Did I update or add new fake data?
- [-] Did I squash my migration?
- [x] Are my changes [backward-compatible](https://github.com/mozilla/foundation.mozilla.org/blob/main/docs/workflow.md#django-migrations-what-to-do-when-working-on-backward-incompatible-migrations). If not, did I schedule a deploy with the rest of the team?

**Documentation:**
- [-] Is my code documented?
- [-] Did I update the READMEs or wagtail documentation?

**Merge Method**
**💡❗Remember to use squash merge when merging non-feature branches into `main`**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -163,7 +163,6 @@ ignore="H017"
 "network-api/networkapi/wagtailpages/templates/wagtailpages/blog_author_index_page.html" = "H006,T003,T032"
 "network-api/networkapi/wagtailpages/templates/wagtailpages/blog_index_page.html" = "H014,T003"
 "network-api/networkapi/wagtailpages/templates/wagtailpages/blog_index_search.html" = "T003,T032"
-"network-api/networkapi/wagtailpages/templates/wagtailpages/blog_page.html" = "T003"
```
Comment (tbrlpld) [inline pyproject.toml:166]: Fixed all template linting issues on the page (added the endblock names)

File: network-api/networkapi/wagtailpages/templates/wagtailpages/fragments/share-buttons.html
Comment (tbrlpld) [inline network-api/networkapi/wagtailpages/templates/wagtailpages/fragments/share-buttons.html:1]: The two `share-button.html` template were identical (only that the one in the buyer's guide directory was already fixed - no spaces in URLs). 

This is now consolidated to be one single template.

File: network-api/networkapi/templates/fragments/share-buttons.html
Comment (tbrlpld) [inline network-api/networkapi/templates/fragments/share-buttons.html:1]: Moved to make clear that this is not specific to the buyersguide.

**Review by danielfmiranda — APPROVED** [review]
LGTM!

**Pull request merged**